src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -120,8 +120,11 @@ case class WorkflowActor(workflow: WorkflowDescriptor,
   private def startRunnableCalls(): State = {
     tryStartingRunnableCalls() match {
       case Success(entries) =>
-        if (entries.nonEmpty) tryStartingRunnableCalls()
-        goto(WorkflowRunning)
+        if (entries.nonEmpty) {
+          startRunnableCalls()
```

Comment:
This is now recursive but previously was not.  I believe the thinking behind the original design was to try starting runnable calls twice: once for the case of a scatter that might need to be exploded, and a second time if there were any exploded shards.  Now it seems `startRunnableCalls` might be called arbitrarily many times.